pkg/common/util/webhooks.go

```diff
@@ -0,0 +1,36 @@
+package util
+
+import (
+	"fmt"
+
+	v1 "github.com/kubeflow/training-operator/pkg/apis/kubeflow.org/v1"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+var supportedJobControllers = sets.New(
+	v1.MultiKueueController,
+	v1.KubeflowJobsController)
+
+func ValidateManagedBy(oldRunPolicy *v1.RunPolicy, newRunPolicy *v1.RunPolicy) field.ErrorList {
+	errs := field.ErrorList{}
+	// Validate immutability
+	if oldRunPolicy != nil && newRunPolicy != nil {
```

Comment:
WDYT @tenzen-y @andreyvelich ? It would also be more consistent with `Job` and `JobSet` placement of the field.

Reply:
Except for a fact that MnagedBy will not be in the same place, but in each spec type I see no issues so far.
The change should be pretty smooth

Reply:
Ah, I see, I got confused, because the code here checks if `runPolicy` is nil, where it cannot be nil, as this is a non-pointer field in spec. 

In that case, we can keep it in spec, and keep the validation code simple (without the runPolicy checks). 

To achieve that I think we need to have two validation functions: one for a new Job (fired on create and update), and validate Job update (fired only on update).

Reply:
Ok, we found the solution. Closing